File: app/src/main/java/com/example/mvctutorial/designpattern/_10composite/Folder.java
package com.example.mvctutorial.designpattern._10composite;

import java.util.ArrayList;
import java.util.List;

public class Folder extends Component {
    List<Component> child = new ArrayList();

    public Folder(String name) {
        super(name);
    }

    Boolean addComponent(Component component) {
        return child.add(component);
    }

    public List<Component> getChildren() {
        return child;
    }
}

File: app/src/main/java/com/example/mvctutorial/designpattern/_10composite/Folder.kt
package com.example.mvctutorial.designpattern._10composite

import java.util.ArrayList

class Folder(name: String): Component(name) {

    private var child = ArrayList<Component>()

    fun addComponent(component: Component): Boolean{
        return child.add(component)
    }

    fun getChildren(): List<Component> {
        return child
    }
}
